tests/testthat/test-check-all.R

```diff
@@ -0,0 +1,186 @@
+context("test-check-all.R")
+
+library("tibble")
+syn <- attempt_instantiate()
+attempt_login(syn)
+annots <- get_synapse_annotations(syn = syn)
+
+test_that("check_all() returns a list of check conditions", {
+  data <- tibble::tibble(
+    metadata_type = c(
+      "manifest",
+      "individual",
+      "biospecimen",
+      "assay"
+    ),
+    file_name = c("file1", "file2", "file3", "file4"),
+    species = "human",
+    assay = "rnaSeq",
+    file_data = c(
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3)))
+    )
+  )
+  res <- check_all(data, annots, syn)
+  expect_equal(class(res), "list")
+  expect_true(all(unlist(
+    purrr::map(
+      res,
+      function(x) {
+        inherits(x, "check_fail") | inherits(x, "check_pass") | inherits(x, "check_warn") # nolint
+      }
+    )
+  )))
+})
+
+test_that("check_all() returns NULL for checks with missing data", {
+  data1 <- tibble::tibble(
+    metadata_type = c(
+      "manifest",
+      "individual",
+      "biospecimen",
+      "assay"
+    ),
+    file_name = c(NA, NA, NA, NA),
+    species = "human",
+    assay = "rnaSeq",
+    file_data = c(
+      list(NULL),
+      list(NULL),
+      list(NULL),
+      list(NULL)
+    )
+  )
+  data2 <- tibble::tibble(
+    metadata_type = c(
+      "manifest",
+      "individual",
+      "biospecimen",
+      "assay"
+    ),
+    file_name = c("file1", NA, NA, NA),
+    species = "human",
+    assay = "rnaSeq",
+    file_data = c(
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(NULL),
+      list(NULL),
+      list(NULL)
+    )
+  )
+  data3 <- tibble::tibble(
+    metadata_type = c(
+      "manifest",
+      "individual",
+      "biospecimen",
+      "assay"
+    ),
+    file_name = c(NA, "file2", NA, "file4"),
+    species = "human",
+    assay = "rnaSeq",
+    file_data = c(
+      list(NULL),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(NULL),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3)))
+    )
+  )
+  data4 <- tibble::tibble(
+    metadata_type = c(
+      "manifest",
+      "individual",
+      "biospecimen",
+      "assay"
+    ),
+    file_name = c("file1", "file2", "file3", NA),
+    species = "human",
+    assay = "rnaSeq",
+    file_data = c(
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+      list(NULL)
+    )
+  )
+  res1 <- check_all(data1, annots, syn)
+  res2 <- check_all(data2, annots, syn)
+  res3 <- check_all(data3, annots, syn)
+  res4 <- check_all(data4, annots, syn)
+
+  expect_true(all(unlist(
+    purrr::map(res1, function(x) {
+      is.null(x)
+    })
+  )))
```

Comment:
`purrr::map_lgl(res1, ~ !is.null(.x))` I think would be sufficient here to replace 

```
unlist(
    purrr::map(res1, function(x) {
      is.null(x)
    })
  )
```